tests/test_rocm_sanity.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+import subprocess
+import re
+from pathlib import Path
+
+THIS_DIR = Path(__file__).resolve().parent
+
+
+def run_command(command):
+    process = subprocess.run(command, capture_output=True, check=False)
+    if process.returncode != 0:
+        pytest.fail(f"The command {command} failed. Failing this test...")
+    return str(process.stdout)
+
+
+class TestROCmSanity:
+    def test_rocminfo(self):
+        output = run_command(["rocminfo"])
+
+        pytest.assume(re.search(r"Device\s*Type:\s*GPU", output))
+        pytest.assume(re.search(r"Name:\s*gfx", output))
+        pytest.assume(re.search(r"Vendor\s*Name:\s*AMD", output))
+
+    def test_clinfo(self):
+        output = run_command(["clinfo"])
+
+        pytest.assume(
+            re.search(r"Device(\s|\\t)*Type:(\s|\\t)*CL_DEVICE_TYPE_GPU", output)
+        )
+        pytest.assume(re.search(r"Name:(\s|\\t)*gfx", output))
+        pytest.assume(
+            re.search(r"Vendor:(\s|\\t)*Advanced Micro Devices, Inc.", output)
+        )
+
+    def test_hip_printf(self):
+        # Compiling .cpp file using hipcc
+        run_command(
+            [
+                "hipcc",
+                f"{str(THIS_DIR)}/hip_printf.cpp",
+                "-o",
+                f"{str(THIS_DIR)}/hip_printf",
```

Comment:
This string/path handling is close but not quite right. See how it runs on Windows:

```python
>>> THIS_DIR = Path("D:\dev\scratch").resolve()
>>> THIS_DIR
WindowsPath('D:/dev/scratch')
>>> f"{str(THIS_DIR)}/hip_printf.cpp"
'D:\\dev\\scratch/hip_printf.cpp'
```

Note the mix of `\\` and `/` there.

When joining paths, do that on `Path` objects, not strings, like so:

```suggestion
                str(THIS_DIR / "hip_printf.cpp"),
                "-o",
                str(THIS_DIR / "hip_printf"),
```